src/persist/src/location.rs

```diff
@@ -86,11 +86,50 @@ impl timely::PartialOrder for SeqNo {
     }
 }
 
-impl Add<u64> for SeqNo {
-    type Output = SeqNo;
+impl SeqNo {
+    /// Returns the next SeqNo in the sequence.
+    pub fn next(self) -> SeqNo {
+        SeqNo(self.0 + 1)
+    }
+}
+
+/// An error coming from an underlying durability system (e.g. s3) or from
+/// invalid data received from one.
+///
+/// TODO: Split this into an enum with variants for failures which mean the
+/// operation _definitely didn't occur_ (e.g. permission denied) vs ones that
+/// _might have occurred_ (e.g. timeout).
+#[derive(Debug)]
+pub struct LocationError {
```

Comment:
LocationError is so awkward! accepting suggestions that do not involve the word "Storage"...